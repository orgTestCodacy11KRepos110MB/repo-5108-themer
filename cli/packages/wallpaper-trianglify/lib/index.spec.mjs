import { colors } from '@themerdev/colors-default';
import { render, renderInstructions } from './index.mjs';
import { describe, expect, it } from 'vitest';

describe('themer "trianglify" wallpaper', () => {
  it(`should return PNG data`, async () => {
    const files = await Promise.all(
      render(colors, { 'themer-wallpaper-trianglify-size': '600x600' }),
    );
    expect(files.length).toBe(4);
    expect(files.filter((file) => /\.png/.test(file.name)).length).toBe(4);
  });
  it('should list output files', async () => {
    const files = await Promise.all(
      render(colors, {
        'themer-wallpaper-trianglify-size': '1000x1000',
        'themer-wallpaper-trianglify-variance': '0.5',
      }),
    );
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
