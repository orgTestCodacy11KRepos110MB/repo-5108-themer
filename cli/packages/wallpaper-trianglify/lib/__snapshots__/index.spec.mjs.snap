// Vitest Snapshot v1

exports[`themer "trianglify" wallpaper > should list output files 1`] = `
"
Files generated:

* \`themer-wallpaper-trianglify-dark-1000x1000-0.5-1.png\`
* \`themer-wallpaper-trianglify-dark-1000x1000-0.5-2.png\`
* \`themer-wallpaper-trianglify-light-1000x1000-0.5-1.png\`
* \`themer-wallpaper-trianglify-light-1000x1000-0.5-2.png\`
"
`;
